financial-templates-lib/ExpiringMultiPartyEventClient.js

```diff
@@ -0,0 +1,167 @@
+        await this._update();
+      } catch (error) {
+        Logger.error({
+          at: "ExpiringMultiPartyEventClient",
+          message: "client polling error",
+          error: error
+        });
+      }
+      await delay(Number(10_000));
+    }
+  };
+
+  _update = async () => {
+    //Events
+    let liquidationBlockQueryNumber =
+      this.liquidationEvents.length > 0 ? this.liquidationEvents[this.liquidationEvents.length - 1].blockNumber + 1 : 0;
+    const liquidationEventsObj = await this.emp.getPastEvents("LiquidationCreated", {
+      fromBlock: liquidationBlockQueryNumber
+    });
+
+    // If there have been previous events retrieve we should only query the chain for newer events.
+    if (liquidationEventsObj.length !== 0) {
+      for (let event of liquidationEventsObj) {
+        this.liquidationEvents.push({
+          transactionHash: event.transactionHash,
+          sponsor: event.returnValues.sponsor,
+          liquidator: event.returnValues.liquidator,
+          liquidationId: event.returnValues.liquidationId,
+          tokensOutstanding: event.returnValues.tokensOutstanding,
+          lockedCollateral: event.returnValues.lockedCollateral,
+          liquidatedCollateral: event.returnValues.liquidatedCollateral
+        });
+      }
+    }
+
+    let disputeBlockQueryNumber =
+      this.disputeEvents.length > 0 ? this.disputeEvents[this.disputeEvents.length - 1].blockNumber + 1 : 0;
+    const disputeEventsObj = await this.emp.getPastEvents("LiquidationDisputed", {
+      fromBlock: disputeBlockQueryNumber
+    });
+    if (disputeEventsObj.length !== 0) {
+      for (let event of disputeEventsObj) {
+        this.disputeEvents.push({
+          transactionHash: event.transactionHash,
+          sponsor: event.returnValues.sponsor,
+          liquidator: event.returnValues.liquidator,
+          disputer: event.returnValues.disputer,
+          liquidationId: event.returnValues.liquidationId,
+          disputeBondAmount: event.returnValues.disputeBondAmount
+        });
+      }
+    }
+
+    let disputeSettlementBlockQueryNumber =
+      this.disputeSettlementEvents.length > 0
+        ? this.disputeSettlementEvents[this.disputeSettlementEvents.length - 1].blockNumber + 1
+        : 0;
+    const disputeSettlementEventsObj = await this.emp.getPastEvents("DisputeSettled", {
+      fromBlock: disputeSettlementBlockQueryNumber
+    });
+    if (disputeSettlementEventsObj.length !== 0) {
+      for (let event of disputeSettlementEventsObj) {
+        this.disputeSettlementEvents.push({
+          transactionHash: event.transactionHash,
+          caller: event.returnValues.caller,
+          sponsor: event.returnValues.sponsor,
+          liquidator: event.returnValues.liquidator,
+          disputer: event.returnValues.disputer,
+          liquidationId: event.returnValues.liquidationId,
+          disputeSucceeded: event.returnValues.DisputeSucceeded
+        });
+      }
+    }
+
+    Logger.debug({
```

Comment:
Same comment as in EMPClient.js, I believe this log is redundant to the one emitted in `update()`

Reply:
ye, you are right actually. Have removed it.